pkg/controller/certificatesigningrequests/util/signername.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+)
+
+type SignerIssuerRef struct {
+	Namespace, Name string
+	Type, Group     string
+}
+
+// SignerIssuerRefFromSignerName will return a SignerIssuerRef from a
+// CertificateSigningRequests.SignerName
+func SignerIssuerRefFromSignerName(name string) (SignerIssuerRef, bool) {
+	split := strings.Split(name, "/")
+	if len(split) != 2 {
+		return SignerIssuerRef{}, false
+	}
+
+	signerTypeSplit := strings.SplitN(split[0], ".", 2)
+	signerNameSplit := strings.Split(split[1], ".")
```

Comment:
suggestion(important): If I understand correctly, I think this function is important as we're defining how cert-manger issuers can be referenced as signers, and I think we're saying that to specify an Issuer, we want the format `foo.bar/namespace.issuerName`.

I think this function would return false with the following (valid) issuer:

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: mynamespace
spec:
  finalizers:
  - kubernetes
apiVersion: v1

---

apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: selfsigned.issuer
  namespace: mynamespace
spec:
  selfSigned: {}
```

I think this would be specified as `foo.bar/mynamespace.selfsigned.issuer`;  `.` is a valid character in an issuer name (but not in the name for a namespace).

Should the switch statement here instead be something like:

```go
if len(signerNameSplit) == 1 {
	return SignerIssuerRef{
    		Namespace: "",
    		Name:      signerNameSplit[0],
    		Type:      signerTypeSplit[0],
    		Group:     signerTypeSplit[1],
	}, true
}

return SignerIssuerRef{
   	Namespace: signerNameSplit[0],
    	Name:      strings.Join(signerNameSplit[1:], "."),
   	Type:      signerTypeSplit[0],
   	Group:     signerTypeSplit[1],
}, true
```

Reply:
this was by far the most important thing in the review, so good to see it's addressed.